source/server/server.cc

```diff
@@ -385,6 +385,10 @@ void InstanceImpl::initialize(const Options& options,
     }
   }
   server_stats_->version_.set(version_int);
+  const std::string fips_ssl_version = "BoringSSL-FIPS";
+  if (VersionInfo::sslVersion() == fips_ssl_version) {
```

Comment:
Instead of string-matching, please add a new function to `VersionInfo` for `sslFipsCompliant` that returns bool, and is set via `#ifdef BORINGSSL_FIPS`.

Reply:
+1, this will leave the openssl fork to be able to indicate the FIPS compliance. cc @envoyproxy/openssl-dev

Reply:
added a new function as suggested.